src/Core/GoAspectContainer.php

```diff
@@ -88,19 +89,35 @@ public function __construct()
             );
         });
 
-        $this->share('aspect.annotation.reader', function(Container $container) {
+        $this->share('aspect.annotation.cache', function(Container $container) {
```

Comment:
Ok, so moving to container - agree, much better. Defined new service, `aspect.annotation.cache`, so reader can get that service for cache engine for annotations. 

Much better, thanks.